### ISSUE TITLE
Auto-correcting `Style/RedundantCondition` can introduce a `Lint/RequireParentheses` offense that changes the meaning of the code

### ISSUE THREAD

**Issue by patbl** [rubocop/rubocop#14051]
Disclaimer: I'm not sure whether this is actually a bug, or if I'm misunderstanding `--autocorrect`'s level of safety (see also https://github.com/rubocop/rubocop/issues/13951).

## Expected behavior

I expect `--autocorrect` not to change what this code does:

```ruby
# frozen_string_literal: true

class Luck
  def bad?(arg)
    [13].include?(arg)
  end
end

def foo
  luck = Luck.new
  foo = false
  if luck.bad? foo
    true
  else
    13
  end
end

p foo # => 13
```

## Actual behavior

`bundle exec rubocop --autocorrect --disable-uncorrectable` changes the code to this:

```ruby
# frozen_string_literal: true

class Luck # rubocop:todo Style/Documentation
  def bad?(arg)
    [13].include?(arg)
  end
end

def foo
  luck = Luck.new
  foo = false
  luck.bad? foo || 13 # rubocop:todo Lint/RequireParentheses
end

p foo # => true
```

## Steps to reproduce the problem

See above.

## RuboCop version

```
1.75.1 (using Parser 3.3.7.3, Prism 1.4.0, rubocop-ast 1.43.0, analyzing as Ruby 3.4, running on ruby 3.4.1) [x86_64-darwin24]
```

**Comment by koic** [comment]
This is a bug in autocorrect. I've opened #14051 to address it.
